src/Development/Shake/Internal/Derived.hs

```diff
@@ -72,22 +75,30 @@ addShakeExtra x = Map.insert (typeOf x) (toDyn x)
 
 -- | @copyFile' old new@ copies the existing file from @old@ to @new@.
 --   The @old@ file will be tracked as a dependency.
+--   Also creates the new directory if necessary.
 copyFile' :: FilePath -> FilePath -> Action ()
 copyFile' old new = do
     need [old]
     putLoud $ "Copying from " ++ old ++ " to " ++ new
+    createDirIfMissing new
```

Comment:
`createDirIfMissing b`
`copyfile a b`
I haven't changed the copyfile which stills copy a to b. I added a check prior to the copyfile to ensure that the directory of b exists, if it doesn't it creates it.
No worries, all comments are welcome.